server/aws-lsp-codewhisperer/src/language-server/agenticChat/context/memorybank/memoryBankController.ts

```diff
@@ -0,0 +1,844 @@
+                }
+
+                this.features.logging.info(
+                    `Memory Bank: parsed ${rankedFilesList.length} ranked files from LLM response`
+                )
+            } catch (error) {
+                this.features.logging.warn(
+                    `Memory Bank: failed to parse LLM ranking response, using TF-IDF fallback: ${error}`
+                )
+                rankedFilesList = analysisResults.rankedFilesList.slice(0, 20)
+            }
+
+            this.features.logging.info(
+                `Memory Bank: using ${rankedFilesList.length} files for documentation generation`
+            )
+
+            // Step 5: Create the comprehensive prompt with ranked files
+            const finalPrompt = MemoryBankPrompts.getCompleteMemoryBankPrompt(rankedFilesList)
+            return finalPrompt
+        } catch (error) {
+            this.features.logging.error(`Memory Bank preparation failed: ${error}`)
+            throw error
+        }
+    }
+
+    /**
+     * Clean and recreate memory bank directory
+     */
+    async cleanMemoryBankDirectory(workspaceFolderUri: string): Promise<void> {
+        try {
+            const memoryBankPath = `${workspaceFolderUri}/${MEMORY_BANK_DIRECTORY}`
+
+            // Remove all existing memory bank files to ensure clean recreation
+            const filesToRemove = ['product.md', 'structure.md', 'tech.md', 'guidelines.md']
+            let removedCount = 0
+            for (const fileName of filesToRemove) {
+                const filePath = `${memoryBankPath}/${fileName}`
+                try {
+                    const exists = await this.features.workspace.fs.exists(filePath)
+                    if (exists) {
+                        await this.features.workspace.fs.rm(filePath)
+                        removedCount++
+                    }
+                } catch (error) {
+                    // Ignore errors when removing files that don't exist
+                    this.features.logging.debug(`Could not remove ${fileName}: ${error}`)
+                }
+            }
+
+            if (removedCount > 0) {
+                this.features.logging.info(`Memory Bank: cleaned ${removedCount} existing files`)
+            }
+
+            // Create the directory structure using mkdir with recursive option
+            await this.features.workspace.fs.mkdir(memoryBankPath, { recursive: true })
+        } catch (error) {
+            this.features.logging.error(`Memory Bank directory creation failed: ${error}`)
+            throw error
+        }
+    }
+
+    /**
+     * files discovery
+     */
+    async discoverAllSourceFiles(
+        workspaceFolderUri: string,
+        extensions: string[]
+    ): Promise<Array<{ path: string; size: number }>> {
+        try {
+            // Recursively discover all source files
+            const sourceFiles = await this.discoverSourceFiles(workspaceFolderUri, extensions)
```

Comment:
Do we need to add limits on file discovery? Without limits, large repos with 10k+ files will create massive vocabularies